crates/anvil/server/src/handler.rs

```diff
@@ -5,26 +5,26 @@ use anvil_rpc::{
     response::{Response, RpcResponse},
 };
 use axum::{
-    extract::{rejection::JsonRejection, Extension},
+    extract::{rejection::JsonRejection, State},
     Json,
 };
 use futures::{future, FutureExt};
 
-/// Handles incoming JSON-RPC Request
-pub async fn handle<Handler: RpcHandler>(
+/// Handles incoming JSON-RPC Request.
+// NOTE: `State` must come first because the `request` extractor consumes the request body.
```

Comment:
ah I remember struggling to make this work
very nice

Reply:
Yeah it took me an hour to figure it out :D
That is literally the only thing that stopped it from compiling, the error being as generic as possible (`fn(..)->.. doesn't implement Service`) and the `debug_handler` macro not working with generics didn't help